packages/plugins/plugin-device-token/README.md

```diff
@@ -0,0 +1,99 @@
+# @segment/analytics-react-native-plugin-device-token
+
+`EnrichmentPlugin` to collect device token values with (Firebase Cloud Messaging)[]. This plugin makes it possible to collect Android's FCM and Apple's APNS device tokens. 
+## Installation
+
+Install the dependencies. 
+
+Using NPM:
+```bash
+npm install --save @segment/analytics-react-native-plugin-device-token
+@react-native-firebase/app @react-native-firebase/messaging
+```
+
+Using Yarn:
+```bash
+yarn add @segment/analytics-react-native-plugin-device-token
+@react-native-firebase/app @react-native-firebase/messaging
+```
+
+Run `pod install` after the installation to autolink the Firebase SDK.
+
+> warning ""
+> Refer to Apple's [Push Notification Services](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns) and [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging) for additional setup requirements. 
+
+## Usage 
+
+
+Follow the [instructions for adding plugins](https://github.com/segmentio/analytics-react-native#adding-plugins) on the main Analytics client:
+
+In your code where you initialize the analytics client call the `.add(plugin)` method with an `DeviceTokenPlugin` instance. 
+
+
+```ts
+import { createClient } from '@segment/analytics-react-native';
+
+import { FirebasePlugin } from '@segment/analytics-react-native-plugin-device-token';
+
+const segmentClient = createClient({
+  writeKey: 'SEGMENT_KEY'
+});
+
+segmentClient.add({ plugin: new DeviceTokenPlugin() });
+```
+
+### updatePermission()
+
+iOS builds require a user to give explicit permission to retrieve the APNS value. This plugin only checks to see if permission has been authorized. You will need to handle permission requests yourself. Once permission has been granted you can call the `updatePermission()` method to begin collecting the device token. 
```

Comment:
It is the same on Android right? We don't ask permissions so you should call the same method when you want to "refresh". Maybe we can remove the iOS part here.

Reply:
no we check the permission status on `configure()` regardless and if the build is android the firebase sdk automatically returns the authorized status.

Reply:
yea, what I mean is this phrase:
`iOS builds require a user to give explicit permission to retrieve the APNS value.`
It is the same for Android, right? They have to give permissions and we don't request them in this library, just check. So both iOS and Android should call `updatePermission()` after they request them